specs/dialog/dialog.md

```diff
@@ -0,0 +1,113 @@
+
+### Prior Art/Examples
+- [FAST-DNA (React)](https://explore.fast.design/components/dialog)
+- [Material UI](https://material-ui.com/components/dialogs/)
+- [Lightning Design](https://www.lightningdesignsystem.com/components/modals/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/modal/code/)
+- [Ant Design](https://ant.design/components/modal/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/modal-dialog)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/dialog)
+- [Windows (UWP)](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/dialogs-and-flyouts/dialogs)
+
+---
+
+## Design
+
+### API
+
+*Component name:*
+- `fast-dialog`
+
+*Attributes:*
+- `modal` - boolean value for modal functionality. Sets aria-modal to true.
+- `visible` - boolean value to toggle the visibility of the dialog
+- `aria-labelledby` - optional based on implementation**
+- `aria-describedby` - optional based on implementation**
+- `aria-label` - optional, based on implementation**
+
+** See the [W3C Specification](https://w3c.github.io/aria-practices/#dialog_roles_states_props) for requirements and details.
+
+*Custom properties:*
+- width
+- height
+
+*Slots:*
+- default slot for content
+
+*Events*
+- dismiss - event fired when a user presses escape or clicks outside the dialog content
+
+### Anatomy and Appearance
+**Structure:**
+```
+<div class="root">
+    <div class="positioning-region">
+        <div class="overlay">
+        <div
+            role="dialog"
+            aria-modal="true"
+            tabindex="-1"
+            class="content-region"
+            aria-describedby="foo"
+            aria-labelledby="bar"
+            aria-label="baz"
+        >
+            <slot></slot>
+        </div>
+    </div>
+</div>
+```
+
+**Appearance:**
+| State | Image |
+| ----- | ----- |
+| default | ![](./images/dialog.png) |
+| dark mode | ![](./images/dialog-dark.png) |
+
+Parts:
+- root - the root of the dialog
+- positioning-region - ensuring the dialog is centered correctly requires a certain structure. Centering with absolute positioning can cause blurry content within the dialg itself. After investigating several implementations, the best and most common way of ensuring the dialog can be centered is to include a div to position the actual dialog itself using something like flexbox or css-grid. This ensures that the blurry content issue does not happen and the dialog centers easily within the screen.
+- modal-overlay - the modal overlay
+- content-region - the region where content is actually rendered. This part is where the `role="dialog"` will actually exist
+
+---
```

Comment:
Is it worth talking a bit about how animation might work for those who want to animate their dialogs in and out of view?

Reply:
Do we want configurable animations if our initial work is to provide the MSFT implementation? I think the initial thinking from design was to bake the animation in here.

Reply:
My working assumption is that animation is something that will be considered part of the "adaptive ui" design system at some point which will provide configurability. If someone wants something completely different than can always create their own instance.

Reply:
Just worth mentioning in the spec I think. Just take your comments and put them in the doc for reference.

I imagine that configurability wouldn't be too hard though. For example, if we put a part attribute on the overlay and a part attribute on the dialog and then we internally apply classes for enter/leave, that would allow the consumer to target those parts and provide their own enter/leave animations. There's a bit more to it, but something to think about.

If we build the animation in, we can do that as a separate set of `ElementStyles` as well, so that consumers could easily re-compose a dialog using all our styles but replacing the animation piece without having to re-write anything.

Just thinking...

Reply:
Sure

Reply:
updated with some additional details